pyaedt/modules/solutions.py

```diff
@@ -1438,743 +1357,647 @@ def assign_weight(self, a, b, taper="flat", port_cont=0):
 
-        phase_shift_A_rad2 = -1 * (
-            (self.Ax * k * math.sin(theta_scan2) * math.cos(phi_scan2))
-            + (self.Ay * k * math.sin(theta_scan2) * math.sin(phi_scan2))
-        )
-        phase_shift_B_rad2 = -1 * (
-            (self.Bx * k * math.sin(theta_scan2) * math.cos(phi_scan2))
-            + (self.By * k * math.sin(theta_scan2) * math.sin(phi_scan2))
-        )
+        Examples
+        --------
+        >>> import pyaedt
+        >>> app = pyaedt.Hfss(specified_version="2023.2", designname="Antenna")
+        >>> setup_name = "Setup1 : LastAdaptive"
+        >>> frequencies = [77e9]
+        >>> sphere = "3D"
+        >>> data = app.get_antenna_ffd_solution_data(frequencies, setup_name, sphere)
+        >>> data.plot_2d_cut(theta_scan=20)
 
-        w_dict = {}
-        w_dict_ang = {}
-        w_dict_mag = {}
-        array_positions = {}
-        port_count = 0
-        for port_name in self.all_port_names:
-            index_str = self.get_array_index(port_name)
-            a = index_str[0]
-            b = index_str[1]
-            w_mag1 = np.round(np.abs(self.assign_weight(a, b, taper=self.taper, port_cont=port_count)), 3)
-            w_ang1 = a * phase_shift_A_rad1 + b * phase_shift_B_rad1
+        """
 
-            w_mag2 = np.round(np.abs(self.assign_weight(a, b, taper=self.taper, port_cont=port_count)), 3)
-            w_ang2 = a * phase_shift_A_rad2 + b * phase_shift_B_rad2
+        if "convert_to_db" in kwargs:  # pragma: no cover
+            self.logger.warning("`convert_to_db` is deprecated since v0.7.5. Use `format_quantity` instead.")
+            if kwargs["convert_to_db"]:
+                format_quantity = "dB10"
+            else:
+                format_quantity = "abs"
+        if "qty_str" in kwargs:  # pragma: no cover
+            self.logger.warning("`qty_str` is deprecated since v0.7.5. Use `farfield_quantity` instead.")
+            farfield_quantity = kwargs["qty_str"]
+
+        data = self.combine_farfield(phi_scan, theta_scan)
+        if farfield_quantity not in data:  # pragma: no cover
+            self.logger.error("Far field quantity not available")
+            return False
 
-            w_dict[port_name] = np.sqrt(w_mag1) * np.exp(1j * w_ang1) + np.sqrt(w_mag2) * np.exp(1j * w_ang2)
-            w_dict_ang[port_name] = np.angle(w_dict[port_name])
-            w_dict_mag[port_name] = np.abs(w_dict[port_name])
+        data_to_plot = data[farfield_quantity]
 
-            array_positions[port_name] = self.element_location(a, b)
-            port_count += 1
+        curves = []
+        if primary_sweep == "phi":
+            x_key = "Phi"
+            y_key = "Theta"
+        else:
+            y_key = "Phi"
+            x_key = "Theta"
+        x = data[x_key]
+        if is_polar:
+            x = [i * 2 * math.pi / 360 for i in x]
+        xlabel = x_key
+        if x_key == "Phi":
+            temp = data_to_plot
+        else:
+            temp = data_to_plot.T
```

Comment:
if primary_sweep == "phi":
            x_key, y_key = "Phi", "Theta"
            temp = data_to_plot
        else:
            y_key, x_key = "Phi", "Theta"
            temp = data_to_plot.T
        x = data[x_key]
        if is_polar:
            x = [i * 2 * math.pi / 360 for i in x]
        xlabel = x_key